### PR TITLE
[WIP] Add contributors to Version model

### PR DESCRIPTION
**Pull request by mvandenburgh**

### CODE REVIEW

File: dandi/publish/models/contributor.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        Validation = "Validation"
+        Visualization = "Visualization"
+        Funder = "Funder"
+        Sponsor = "Sponsor"
+        StudyParticipant = "StudyParticipant"
+        Other = "Other"
+
+    name = models.CharField(default=str, max_length=150) # TODO: verify max_length is correct
+    email = models.TextField(default=str) # TODO: need max length, use charfield if possible
+    orcid = models.TextField(default=str) # TODO: same as email
```
Comment (mvandenburgh) [inline dandi/publish/models/contributor.py:35]: @mgrauer Do we have maximum lengths for contributor `name`, `email` and `orcid`?

File: dandi/publish/models/contributor.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from django.db import models
+from django.contrib.postgres.fields import ArrayField, JSONField
+
+from .role import Role
+
+
+class Contributor(models.Model):
+    # https://github.com/dandi/schema/blob/master/releases/1.0.0-rc1/dandiset.json#L424
+
+    identifier = models.TextField(max_length=65536, blank=True)
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:10]: Omit `identifier`.

File: dandi/publish/models/contributor.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    name = models.CharField(max_length=255)
+
+    email = models.CharField(max_length=254, blank=True)
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:14]: Use https://docs.djangoproject.com/en/3.0/ref/models/fields/#emailfield

File: dandi/publish/models/contributor.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    url = models.TextField(max_length=65536, blank=True)  # TODO: validate length on the api level
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:16]: Use https://docs.djangoproject.com/en/3.0/ref/models/fields/#urlfield

File: dandi/publish/models/contributor.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+from django.db import models
+from django.contrib.postgres.fields import ArrayField, JSONField
+
+from .role import Role
+
+
+class Contributor(models.Model):
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:7]: Should be `ContributorVersion` (or something like that).

File: dandi/publish/models/version.py [hunk 6]
```diff
@@ -33,6 +34,8 @@ class Version(models.Model):
     name = models.CharField(max_length=150)
     description = models.TextField(max_length=3000)
 
+    contributors = models.ManyToManyField(Contributor)
```
Comment (brianhelba) [inline dandi/publish/models/version.py:37]: What about ordering??

File: dandi/publish/models/contributor.py (hunk 3 above)
Comment (brianhelba) [inline dandi/publish/models/contributor.py:14]: Make all of these optional fields `null=True` too.

File: dandi/publish/models/contributor.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    name = models.CharField(max_length=255)
+
+    email = models.CharField(max_length=254, blank=True)
+
+    url = models.TextField(max_length=65536, blank=True)  # TODO: validate length on the api level
+
+    roleName = models.ManyToManyField(Role)
+
+    includeInCitation = models.BooleanField(blank=True)
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:20]: Make `default=True` instead, but not blankable or nullable.

File: dandi/publish/models/contributor.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    awardNumber = models.TextField(max_length=65536, blank=True)  # TODO: validate length
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:22]: `CharField(max_length=64)`

File: dandi/publish/models/contributor.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+    contributorType = models.CharField(max_length=12)  # 'Person' or 'Organization'
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:24]: Cut this, assume it's always a Person.

File: dandi/publish/models/contributor.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+    affiliation = ArrayField(models.CharField(max_length=255), blank=True) # TODO: remove these arrayfields
+
+    contactPoint = ArrayField(models.CharField(max_length=255), blank=True)
```
Comment (brianhelba) [inline dandi/publish/models/contributor.py:28]: Cut this, since it's always a Person for now.

**Pull request closed without merging**